### ISSUE TITLE
Automate process of adding merged PR to milestone and removal of labels

### ISSUE THREAD

**Issue by krmanik** [ankidroid/Anki-Android#10410]
Might be useful to automate this in the longer-term, either with a check to confirm that all merged PRs have a milestone, or a bot reminder to set a milestone once merged

_Originally posted by @david-allison in https://github.com/ankidroid/Anki-Android/issues/10353#issuecomment-1044721187_

**Comment by mikehardy** [comment]
Classic errors on both (same error) when PR closed:

> Resource not accessible by integration

That's because PRs on forks can't run in the full context of the main repository or it could leak information from main repo action context (secrets etc) to untrusted code (from a fork etc).

It was running in the fork/branch context though, so perhaps now that it is on main the next merge will work successfully? Going to reopen the issue until testing proves it works or we may need to give the workflow a special token or something

**Comment by krmanik** [comment]
I will check it on my AnkiDroid fork. Then creates again PR.

**Comment by krmanik** [comment]
I have tested it, it seems to works but I will wait on next PR merge to see if works or not.

I have tested on this 
https://github.com/krmanik/Anki-Android/pull/19

**Comment by mikehardy** [comment]
It works in your repo on PRs from your repo because they are not isolated from getting valid tokens and running actions.
It's the fork case that causes this - it really is a classic for github actions

I think it needs to be `pull_request_target` not `pull_request` 
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target
Like this https://github.com/interledger/rafiki/pull/280/files

**Comment by krmanik** [comment]
I have created PR considering this.  It should work now.

**Comment by mikehardy** [comment]
Leaving open until #10676 results - which seem good! but aren't reflected on the PR for me - are confirmed good.

**Comment by krmanik** [comment]
Can it be the solution?
https://github.community/t/github-actions-bot-labeling-triggers-not-the-issue-labeled-event/16995/2

**Comment by mikehardy** [comment]
No - that one is in the area, but is more about the act of labeling itself then causing a fresh round of workflow triggers to fire (and run dependent workflows)

We don't care about a second round of actions firing off, we just want the actual label and milestone that *appear* (to the github action to be successfully added) to actually be added

**Comment by krmanik** [comment]
I will try again later after getting more info on it.

**Comment by krmanik** [comment]
Let this issue reopened till it is not fixed.
Currently focusing on labeling. Next PR will be related to milestone.

**Comment by mikehardy** [comment]
@krmanik success! https://github.com/ankidroid/Anki-Android/pull/10707 / https://github.com/ankidroid/Anki-Android/runs/5835147220?check_suite_focus=true

**Comment by krmanik** [comment]
It is working. 
Next is milestone.

**Comment by krmanik** [comment]
The scripts to remove labels does not worked on the #10702. 
It is difficult to figure out. Still debugging.

**Comment by mikehardy** [comment]
@krmanik looks like @david-allison manually removed the tag and just beat the bot to it ?

**Comment by mikehardy** [comment]
Okay - here's what I just saw:
![image](https://user-images.githubusercontent.com/782704/161797203-f7e14d24-b071-41d4-8481-b5d16fa49e1b.png)

on #10712 - it was a little weird but I don't think there is an actionable problem, I think we can rely on label removal - I merged it and waited+watched and saw what I pasted as a screen region just above

But the labels list on the PR page still had "Needs Second Approval" on it! It only went away in the label list if I force-reloaded the page, or opened the PR again in a new page.

That's a GitHub web UI thing I think, the label was definitely removed.

So for me: the label removing is working as well as it can now, I think it's good to go.

**Comment by krmanik** [comment]
I have checked it and that's is GitHub web UI issues. The bot removed the labels but UI not updated, it needs to reopen the tab.

**Comment by mikehardy** [comment]
Leaving open until testing confirms milestone additions

**Comment by mikehardy** [comment]
Milestone not quite working as noted in testing https://github.com/ankidroid/Anki-Android/pull/10733#issuecomment-1091980524

**Comment by krmanik** [comment]
I will try again later. I will create new PR.
